packages/core/admin/server/services/passport.js

```diff
@@ -24,7 +24,11 @@ const registerAuthEvents = () => {
   const eventList = Object.entries(events).filter(keyIsValidEventName).filter(valueIsFunctionType);
 
   for (const [eventName, handler] of eventList) {
-    strapi.eventHub.on(authEventsMapper[eventName], handler);
+    strapi.eventHub.addSubscriber((name, ...args) => {
+      if (name === authEventsMapper[eventName]) {
+        handler(...args);
+      }
+    });
```

Comment:
I think ideally the logic to test if the event is the right one would have been inside `handler`, but since handler comes from a user config, that would have been a breaking change